zkvm/src/vm.rs

```diff
@@ -567,6 +570,29 @@ where
         })
     }
 
+    fn delegate(&mut self) -> Result<(), VMError> {
+        let sig = self.pop_item()?.to_data()?.to_bytes();
+        let prog = self.pop_item()?.to_data()?.to_bytes();
+        let contract = self.pop_item()?.to_contract()?;
+
+        let R = SliceReader::parse(&sig[0..32], |r| r.read_point())?;
```

Comment:
this is undoing the safety feature of `SliceReader::parse`: you are not longer checking that `sig` is fully read.

Instead use one `parse` call to guarantee that all bytes are read:

```rust
let (R,s) = SliceReader::parse(&sig, |r| {
    Ok((r.read_point()?, r.read_scalar()?))
})?;
```

Reply:
sorry didn't see this